packages/forklift-console-plugin/src/modules/Plans/views/details/tabs/VirtualMachines/modals/PlanVMsEditModal.tsx

```diff
@@ -0,0 +1,146 @@
+import { Modal, ModalVariant } from '@patternfly/react-core';
+
+import './PlanVMsDeleteModal.style.css';
+import './PlanVMsEditModal.style.css';
+
+export interface PlanVMsEditModalProps {
+  plan: V1beta1Plan;
+  editAction: PlanEditAction;
+}
+
+export const PlanVMsEditModal: React.FC<PlanVMsEditModalProps> = ({ plan, editAction }) => {
+  const { toggleModal } = useModal();
+  const { t } = useForkliftTranslation();
+  const projectName = plan?.metadata?.namespace;
+
+  // Retrieve k8s source provider
+  const [sourceProvider, sourceProviderLoaded, sourceProviderLoadError] =
+    useK8sWatchResource<V1beta1Provider>({
+      groupVersionKind: ProviderModelGroupVersionKind,
+      namespaced: true,
+      name: plan?.spec?.provider?.source?.name,
+      namespace: plan?.spec?.provider?.source?.namespace,
+    });
+
+  // Retrieve k8s target provider
+  const [targetProvider, targetProviderLoaded, targetProviderLoadError] =
+    useK8sWatchResource<V1beta1Provider>({
+      groupVersionKind: ProviderModelGroupVersionKind,
+      namespaced: true,
+      name: plan?.spec?.provider?.destination?.name,
+      namespace: plan?.spec?.provider?.destination?.namespace,
+    });
+
+  const [providers, providersLoaded, providersLoadError] = useK8sWatchResource<V1beta1Provider[]>({
+    groupVersionKind: ProviderModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace: plan?.metadata?.namespace,
+  });
+
+  // Retrieve all k8s Network Mappings
+  const [networkMaps, networkMapsLoaded, networkMapsError] = useK8sWatchResource<
+    V1beta1NetworkMap[]
+  >({
+    groupVersionKind: NetworkMapModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace: plan?.metadata?.namespace,
+  });
+
+  // Retrieve all k8s Storage Mappings
+  const [storageMaps, storageMapsLoaded, storageMapsError] = useK8sWatchResource<
+    V1beta1StorageMap[]
+  >({
+    groupVersionKind: StorageMapModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+    namespace: plan?.metadata?.namespace,
+  });
+
+  const [vmData] = useInventoryVms(
+    { provider: sourceProvider },
+    sourceProviderLoaded,
+    sourceProviderLoadError,
+  );
+  const initialSelectedIds = plan.spec.vms.map((specVm) => specVm.id);
+  const selectedVMs = vmData.filter((vm) => initialSelectedIds.includes(vm.vm.id));
+
+  const planNetworkMaps = networkMaps
+    ? networkMaps.find((net) => net?.metadata?.name === plan?.spec?.map?.network?.name)
+    : null;
+  const planStorageMaps = storageMaps
+    ? storageMaps.find((storage) => storage?.metadata?.name === plan.spec.map?.storage?.name)
+    : null;
+
+  const finishedLoading =
+    providersLoaded &&
+    sourceProviderLoaded &&
+    targetProviderLoaded &&
+    networkMapsLoaded &&
+    storageMapsLoaded &&
+    vmData.length > 0;
+  const hasErrors =
+    providersLoadError ||
+    sourceProviderLoadError ||
+    targetProviderLoadError ||
+    networkMapsError ||
+    storageMapsError;
```

Comment:
please useMemo

Reply:
no expensive computation here, so I'd say useMemo would be overkill